### DISCUSSION TITLE
Incorrect "github repo info" in header

### DISCUSSION THREAD

**Discussion by buhtz** [squidfunk/mkdocs-material#7223] [Q&A] [answered]
![image](https://github.com/squidfunk/mkdocs-material/assets/11861496/788d7540-8f88-43ac-81b9-23678699fd7b)

That field in the header indicate tag "v1.4.3". From a users perspective I would understand that the docu I do see currently is related to that version/tag and build from it.

But that is not true. The docu generating that field is far beyond (~100 commits) that tag.

So, how is this field generated? Can I manipulate it somehow? Can I deactivate it?

Is there a way to indicate for real on which version (branch name, last commit, ...) that docu was build on?

**Comment by squidfunk** [comment]
I already answered this [here](https://github.com/mkdocs/mkdocs/discussions/3719#discussioncomment-9590492).

> The information is automatically extracted from the GitHub repository set in [`repo_url`](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/#repository). If you set the URL to the repository of your documentation repository, the version from the latest release of this repository will be used, if any release exists. If no release exists, there will be no version tag.
> 
> Also see https://github.com/squidfunk/mkdocs-material/issues/2082#issuecomment-1790423790.

**Reply by buhtz** [reply]
Can I somehow (without using CSS hacks) deactivate the whole element?

Sidequestion: Is the content in this element static or are there requests to microsoft github everytime one user opens the documentation?

**Reply by kamilkrzyskow** [reply, marked as answer]
> Can I somehow (without using CSS hacks) deactivate the whole element?

The HTML is included based on the `repo_url` configuration setting in your mkdocs.yml file:
https://www.mkdocs.org/user-guide/configuration/#repo_url

It's added in both the `nav` and `header`:
https://github.com/squidfunk/mkdocs-material/blob/2412a1b7bc0713730ec3d134b26fe503d9421c41/src/templates/partials/nav.html#L55-L60
https://github.com/squidfunk/mkdocs-material/blob/2412a1b7bc0713730ec3d134b26fe503d9421c41/src/templates/partials/header.html#L103-L109

So you can disable it by not setting the `config.repo_url` or setting it to an empty string.

You can also disable it via overrides https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme
and replace the included `source.html` partial with an empty version of the file. However, the wrapper will still be included, to exclude the wrapper overriding the `nav.html` and `header.html` templates would be required... or better use extra CSS.

> Is the content in this element static or are there requests to microsoft github everytime one user opens the documentation?

The fetch to the GitHub API is done once per browser session, so it fetches data once and caches it until the users restarts their browser.

---

EDIT: I see that you're a Codeberg user, if I remember correctly https://github.com/Andre601 had some extra javascript to fetch that statistic data from Codeberg, but I can't find it right now in the discussions 🤔 If you're interested, perhaps ping him?

**Reply by squidfunk** [reply]
> The fetch to the GitHub API is done once per browser session, so it fetches data once and caches it until the users restarts their browser.

Correct. Note that it respects the [cookie consent](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#cookie-consent), if configured.

**Reply by buhtz** [reply]
> The fetch to the GitHub API is done once per browser session, so it fetches data once and caches it until the users restarts their browser.

I would treat that as heavy bug. Most users understand MkDocs and its themes as a static site generator.

It might be that I always do not understand the intention of that git-version-stars element in the header. But it doesn't count as a version information. I will check out other solutions to provide a correct version information to my users without exposing them to a US company treating humans as products and disrespecting their privacy.

**Reply by squidfunk** [reply]
It's not a bug. Most users find this very convenient. But I understand that preferences might be different.

No worries, just remove `repo_url` and nothing will be queried or displayed at all. Additionally, the fully configurable cookie consent allows you to deactivate it by default, so the user needs to opt-in ☺️ If you wish to keep the repository information, override the [`source.html`](https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/source.html) partial and remove `data-md-component` attribute.

That's three levels of possible configuration/customization to fit basically any taste.

**Reply by pacharanero** [reply]
This issue did catch me out, the new Release was not being updated and I couldn't work out where it was being persisted. It was fixed by deleting the cached browser storage manually in the Developer Console. F5 cache clearing didn't work.
